### PR TITLE
Disable CSRF protection via config switch

### PR DESCRIPTION
**Pull request by DeepDiver1975**
## Description
In some rare setups CSRF protection is done on the firewall level - e.g. F5 ASM

## Motivation and Context
No longer need to manually patch installations.

## Types of changes
- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:
- [x] My code follows the code style of this project.
- [x] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have read the **CONTRIBUTING** document.
- [ ] I have added tests to cover my changes.
- [ ] All new and existing tests passed.

### CODE REVIEW

File: lib/private/AppFramework/Http/Request.php [hunk 1]
```diff
@@ -455,6 +455,11 @@ public function passesCSRFCheck() {
 			return false;
 		}
 
+		// csrf protection is handled outside of ownCloud
+		if ($this->config->getSystemValue('csrf.disabled', false)) {
```
Comment (PVince81) [inline lib/private/AppFramework/Http/Request.php]: add to config.sample.php ?

**Comment by PVince81** [comment]
Did you check whether this covers all code paths ? Some older ajax APIs might go a different path. But with a bit of luck they all still land here.

**Comment by mmattel** [comment]
Pls also add a documentation PR

**Comment by PVince81** [comment]
@mmattel it should be enough to add it to config.sample.php.
There's a magic script that imports all comemnts from config.sample.php into the documentation.

cc @settermjd

File: lib/private/AppFramework/Http/Request.php (hunk 1 above)
Comment (DeepDiver1975) [inline lib/private/AppFramework/Http/Request.php]: done

**Comment by DeepDiver1975** [comment]
> But with a bit of luck they all still land here.

yes - double checked

**Comment by PVince81** [comment]
:+1:

File: config/config.sample.php [hunk 2]
```diff
@@ -203,6 +203,16 @@
 'token_auth_enforced' => false,
 
 /**
+ * Disable ownCloud's built-in CSRF protection mechanism.
```
Comment (DeepDiver1975) [inline config/config.sample.php]: @settermjd mind double checking this documentation fragment? THX

Comment (PVince81) [inline config/config.sample.php]: @settermjd ping

**Review by settermjd — APPROVED** [review]

**Comment by DeepDiver1975** [comment]
THX @settermjd @PVince81

**Pull request merged**